File: sdkwork-backend-sdk-go/types/plus_card_template_vo.go
package types

// 会员卡模板Value Object
type PlusCardTemplateVO struct {
	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
	Name string `json:"name"`
	TemplateCode string `json:"templateCode"`
	CardType string `json:"cardType"`
	CodeType string `json:"codeType"`
	Title string `json:"title"`
	Description string `json:"description"`
	BrandName string `json:"brandName"`
	LogoUrl string `json:"logoUrl"`
	Notice string `json:"notice"`
	Color string `json:"color"`
	Quantity int `json:"quantity"`
	GetLimit int `json:"getLimit"`
	CanShare bool `json:"canShare"`
	CanGiveFriend bool `json:"canGiveFriend"`
	MinimumBalance float64 `json:"minimumBalance"`
	InitialBalance float64 `json:"initialBalance"`
	DiscountRate float64 `json:"discountRate"`
	ValidityType string `json:"validityType"`
	StartTime string `json:"startTime"`
	EndTime string `json:"endTime"`
	ValidityDays int `json:"validityDays"`
	IsDeleted bool `json:"isDeleted"`
}

File: sdkwork-backend-sdk-typescript/src/types/plus-card-template-vo.ts
/** 会员卡模板Value Object */
export interface PlusCardTemplateVO {
  /** 创建时间 */
  createdAt?: string;
  /** 更新时间 */
  updatedAt?: string;
  /** 模板名称 */
  name?: string;
  /** 模板编码 */
  templateCode?: string;
  /** 卡片类型(枚举) */
  cardType?: 'DEFAULT' | 'MEMBER' | 'GIFT' | 'DISCOUNT' | 'CASH' | 'NORMAL';
  /** 编码类型(枚举) */
  codeType?: 'DEFAULT' | 'TEXT' | 'BARCODE' | 'QRCODE';
  /** 标题 */
  title?: string;
  /** 描述 */
  description?: string;
  /** 品牌名称 */
  brandName?: string;
  /** Logo URL */
  logoUrl?: string;
  /** 公告 */
  notice?: string;
  /** 颜色 */
  color?: string;
  /** 数量 */
  quantity?: number;
  /** 获取限制 */
  getLimit?: number;
  /** 是否可分享 */
  canShare?: boolean;
  /** 是否可赠友 */
  canGiveFriend?: boolean;
  /** 最低余额 */
  minimumBalance?: number;
  /** 初始余额 */
  initialBalance?: number;
  /** 折扣率 */
  discountRate?: number;
  /** 有效期类型 */
  validityType?: 'DEFAULT' | 'FIXED' | 'DAYS_AFTER_RECEIVE' | 'DAYS_AFTER_ACTIVATION';
  /** 开始时间 */
  startTime?: string;
  /** 结束时间 */
  endTime?: string;
  /** 有效天数 */
  validityDays?: number;
  /** 是否删除 */
  isDeleted?: boolean;
}
